File: beecrowd/1012 - Área.go
package main

import (
	"fmt"
	"math"
)

type Area struct {
	a float64
	b float64
	c float64
}

func (area *Area) leitura() {
	fmt.Scan(&area.a, &area.b, &area.c)
}

func (area *Area) triangulo() float64 {
	return area.a * area.c / 2
}

func (area *Area) circulo() float64 {
	const PI = 3.14159

	return math.Pow(area.c, 2) * PI
}

func (area *Area) trapezio() float64 {
	return (area.a + area.b) * area.c / 2
}

func (area *Area) quadrado() float64 {
	return math.Pow(area.b, 2)
}

func (area *Area) retangulo() float64 {
	return area.a * area.b
}

func (area *Area) imprime() {
	fmt.Printf("TRIANGULO: %.3f\n", area.triangulo())
	fmt.Printf("CIRCULO: %.3f\n", area.circulo())
	fmt.Printf("TRAPEZIO: %.3f\n", area.trapezio())
	fmt.Printf("QUADRADO: %.3f\n", area.quadrado())
	fmt.Printf("RETANGULO: %.3f\n", area.retangulo())
}

func main() {

	var area Area

	area.leitura()
	area.imprime()
}

File: beecrowd/1012 - Área.kt
import java.util.*
import kotlin.math.pow

fun areaTrianguloRetangulo(a: Double, c: Double): Double = a * c / 2.0
fun areaCirculo(c: Double): Double = c.pow(2.0) * 3.14159
fun areaTrapezio(a: Double, b: Double, c: Double): Double = (a + b) * c / 2.0
fun areaQuadrado(b: Double): Double = b.pow(2.0)
fun areaRetangulo(a: Double, b: Double): Double = a * b

fun leTresDoubles(): Triple<Double, Double, Double> {
    val linha = readLine()!!
    val elementos = linha.split(" ")
    val primeiro = elementos[0].toDouble()
    val segundo = elementos[1].toDouble()
    val terceiro = elementos[2].toDouble()

    return Triple(primeiro, segundo, terceiro)
}

fun main(args: Array<String>) {

    val (a, b, c) = leTresDoubles()

    println("TRIANGULO: %.3f".format(Locale.US, areaTrianguloRetangulo(a, c)))
    println("CIRCULO: %.3f".format(Locale.US, areaCirculo(c)))
    println("TRAPEZIO: %.3f".format(Locale.US, areaTrapezio(a, b, c)))
    println("QUADRADO: %.3f".format(Locale.US, areaQuadrado(b)))
    println("RETANGULO: %.3f".format(Locale.US, areaRetangulo(a, b)))
}